### PR TITLE
COMPUTE-219 Remove dxpy dependencies requests and cryptography

### PR DESCRIPTION
**Pull request by kpjensen**
Remove both `requests` and `cryptography` from dxpy requirements. 
Fixed pytest warnings and removed some python2 specific code.

### CODE REVIEW

File: src/python/doc/dxpy.rst [hunk 1]
```diff
@@ -16,14 +16,6 @@ This Python 2.7 package includes three key modules:
 
 It has the following external dependencies:
```
Comment (wormsik) [inline src/python/doc/dxpy.rst:18]:
```suggestion
```

Comment (wormsik) [reply src/python/doc/dxpy.rst:18]: plus I would also remove Python 2.7 on line 4

**Review by wormsik — CHANGES_REQUESTED** [review]

File: src/python/requirements.txt [hunk 2]
```diff
@@ -3,9 +3,8 @@ argcomplete>=1.9.4,<2.0.0; python_version < "3.10"
 websocket-client==0.54.0
 python-dateutil>=2.5
 psutil>=5.9.3
-requests>=2.8.0,<2.28; python_version < "3.7"
-requests>=2.8.0,<2.29; python_version >= "3.7"
-cryptography>=3.4.2,<41; python_version > "3.5"
+certifi
+urllib3>=1.26.18,<2.1.0
```
Comment (wormsik) [inline src/python/requirements.txt:7]: why don't we pin the most recent version? is it because of SecureTransport TLS deprecation?

Comment (kpjensen) [reply src/python/requirements.txt]: I have not done extensive testing with `2.1.0` which has many deprecations, but seems okay so far in testing. `urllib3` deprecations on new minor version updates have caused dxpy errors in the past. 
`v1.26.x` is still maintained and `<2` could be required by other packages a user may have installed.

**Review by wormsik — APPROVED** [review]
LGTM 👍

**Pull request merged**